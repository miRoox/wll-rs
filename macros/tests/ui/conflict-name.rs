use wll::{Error, ErrorKind, Result};
use wll_macros::{export, setup, teardown};

#[setup]
fn setup() {}

#[teardown]
fn teardown() {}

#[export(add_two)]
fn add_two(a: isize, b: isize) -> Result<isize> {
    a.checked_add(b)
        .ok_or_else(|| Error::from(ErrorKind::NumericalError))
}

fn main() {}
