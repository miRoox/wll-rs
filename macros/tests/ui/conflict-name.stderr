error[E0428]: the name `add_two` is defined multiple times
  --> $DIR/conflict-name.rs:11:1
   |
10 | #[export(add_two)]
   | ------------------ previous definition of the value `add_two` here
11 | fn add_two(a: isize, b: isize) -> Result<isize> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `add_two` redefined here
   |
   = note: `add_two` must be defined only once in the value namespace of this module
